java/grpc-client/src/test/java/io/client/grpc/GrpcClientStaticAuthTest.java

```diff
@@ -0,0 +1,57 @@
+package io.client.grpc;
+
+import io.vitess.client.Context;
+import io.vitess.client.RpcClientTest;
+import io.vitess.client.grpc.GrpcClientFactory;
+import io.vitess.client.grpc.StaticAuthCredentials;
+import java.net.ServerSocket;
+import java.net.URI;
+import java.util.Arrays;
+import org.joda.time.Duration;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class GrpcClientStaticAuthTest extends RpcClientTest {
+  private static Process vtgateclienttest;
+
+  @BeforeClass
+  public static void setUpBeforeClass() throws Exception {
+    String vtRoot = System.getenv("VTROOT");
+    if (vtRoot == null) {
+      throw new RuntimeException("cannot find env variable VTROOT; make sure to source dev.env");
+    }
+    URI staticAuthFile = GrpcClientStaticAuthTest.class.getResource("grpc_static_auth.json").toURI();
+
+    ServerSocket socket = new ServerSocket(0);
+    int port = socket.getLocalPort();
+    socket.close();
+
+    vtgateclienttest = new ProcessBuilder(Arrays.asList(
+        vtRoot + "/bin/vtgateclienttest",
+        "-logtostderr",
+        "-grpc_port", Integer.toString(port),
+        "-service_map", "grpc-vtgateservice",
+        "-grpc_auth_mode", "static",
+        "-grpc_auth_static_password_file", staticAuthFile.getPath()
+    )).inheritIO().start();
+
+    Context ctx = Context.getDefault().withDeadlineAfter(Duration.millis(5000L));
+    client = new GrpcClientFactory()
+        .setCallCredentials(new StaticAuthCredentials("test-username", "test-password"))
+        .create(ctx, "localhost:" + port);
```

Comment:
For completeness I feel like we should add a test case that demonstrates what happens when the credentials are wrong.

Glancing at the other tests (e.g. `GrpcClientTlsClientAuthTest`) it looks like we've only tested the happy path, but I think it'd be better to also future proof that this is _actually_ doing what we think.

Reply:
👍 added `testWrongPassword` case

Reply:
OK well looking at that test case it seems like the error only comes when you called `client.GetSrvKeyspace` so I think we should add that to _this_ test case to prove that it actually works 😁

Reply:
Not exactly..

In the `testWrongPassword` case it's clear that you needed to call `client.GetSrvKeyspace` in order to trigger the error, since creating the client works fine, but trying to use the client to emit a GRPC is what fails.

I was confused where in the test you were showing that these static auth credentials are actually working as expected since the only thing you do here is create the client, not use it.

Then upon further reading, I realize that all the _actual_ test cases are inherited from RpcClientTest so this is sufficient.